realm/realm-library/src/main/java/io/realm/internal/RealmProxyMediator.java

```diff
@@ -134,7 +134,7 @@
      * After being inserted any changes to the original objects will not be persisted.
      *
      * @param realm reference to the {@link Realm} where the objects will be inserted.
-     * @param objects Collection of {@link RealmObject} to insert or update.
+     * @param objects Collection of {@link RealmObject} to insert or update. This must not be empty.
```

Comment:
I think that "must not" is a bit too strong. If `objects` is empty, it is a no-op (we don't throw an exception). Maybe "should not" is better.